core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -511,6 +523,12 @@ private void scheduleUpdate()
         executor.execute(this::sendUpdate);
     }
 
+    private synchronized void triggerUpdate()
```

Comment:
nit: would be great to add a comment why it needs to be synchronized (so that `needsUpdate` is not cleared in `sendUpdate` before actual request is sent)